doc/api/globals.md

```diff
@@ -560,6 +560,23 @@ changes:
 
 A browser-compatible implementation of {Headers}.
 
+## `localStorage`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental.
+
+A browser-compatible implementation of [`localStorage`][]. Data is stored
+unencrypted in the current working directory, with a storage quota of 10 MB.
```

Comment:
Can the working directory be configured? For example when working with Yarn workspaces, the cwd may be different depending on the package the user is working on, leading to surprising behaviours.

Reply:
Yeah, I agree. Definitely important to be able to set the storage location explicitly. I wouldn't say this is necessary immediately for this to land but I'd consider it to be a criteria for graduating from experimental.

Reply:
(I didn't want to post this globally to not water down the main discussion)

> This seems incorrect to me? nvm switches the binary, but not the working directory.

Sorry for the confusion. Let me clarify: if I install something like `gh` CLI (let's assume it was an npm package) it will be placed in `~/.nvm/versions/node/v18.18.2/bin/gh`. If I update to `v20.8.1` and install `gh` again, the path changes. And if `localStorage` would be associated with the path of Node.js program, the data is lost. Using the working directory doesn't seem to make any sense to me in this context. If `gh auth` would use `localStorage` to store the auth token, why would it matter in which project (working directory) I want to execute `gh` commands? And what if also use `vercel` in the same project and both use `localStorage.getItem('auth')` and don't namespace their keys (why would they)? Why would I pollute every working directory with `*.localstorage` files? Unless I completely misunderstand the use-case of `localStorage`.

Thinking about it, it would be totally fine that switching from `~/.nvm/versions/node/v18.18.2/bin/gh` to `~/.nvm/versions/node/v20.8.1/bin/gh` would use two separate `localStorage`. It might even be desirable to have these complete separate. In this case nvm would even make the backwards compatibility of `localStorage` a non-issue.

TLDR: I think using the path to the Node.js program sounds like a sane identifier for `localStorage`, the current working directory doesn't even sound like a candidate. The SQLite file(s) would then be in a central place somewhere in `~/.node`?

Reply:
I am fine with requiring an explicit CLI flag to specify the location for `localStorage` and only making `sessionStorage` available otherwise. I'm not sure if this is what you meant, @tniessen, but I would not want to make `localStorage` be persistent in some cases and transient in others.

Reply:
I think both should be added. The spec, IIRC, doesn't mention that localStorage *must be* persistent, and browsers don't treat it as such always. Browsers will sometimes clear it if your computer is low on storage, if you're in a private browsing window, and for a whole sleuth of other, almost random, reasons. Browsers are so bad at keeping site's storage that there's a new [storage api](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager) on navigator where you can ask a browser to persist the data.

Reply:
Right, `localStorage` is also not a global storage, so interlacing the localStorage storage to a context of the running working directory could make sense? There should be a way of determining by default where the storage should be stored but something that we definitely should support is being overridden.

Reply:
would process.chdir change the local storage context?

Reply:
I certainly think it is reasonable to gate availability of `localStorage` on the use of an command line flag that explicitly sets the location. That addresses a number of possible concerns and edge cases, I think.